data_providers/quesions_provider.gd

```diff
@@ -14,14 +15,16 @@ var questions = {
 var current_question_index = -1;
 
 func _init():
-	var questions_from_json = load_json_file(file_path)["questions"]
+	var questions_from_json = load_json_file(file_path)
 	assert(questions_from_json != null, "No questions found.")
-	for question in questions_from_json:
+	for i in range(0, len(questions_from_json)):
+		var question = questions_from_json[str(i)]
```

Comment:
This is potentially dangerous to assume that every number will be there. If I'm not mistaken it should be 
```gdscript 
for i in len(questions_from_json)
```
anyways in case it is a safe assumption which makes the whole id pointless imo.